File: utility/src/main/java/com/ellevsoft/utility/LogUtils.java
package com.ellevsoft.utility;

import android.util.Log;

public class LogUtils {

    private final static String LOGTAG = "LogUtils";

    // Shows logs on debug version only (Does not appear on the release version)
    private final static boolean DEBUG = BuildConfig.DEBUG;

    public static void v(String message, Object... args) {
        if (DEBUG) {
            Log.v(LOGTAG, args == null || args.length == 0
                    ? message : String.format(message, args));
        }
    }

    public static void v(String tag, String message, Object... args) {
        if (DEBUG) {
            Log.v(LOGTAG + "/" + tag, args == null || args.length == 0 ? message
                    : String.format(message, args));
        }
    }

    public static void d(String message, Object... args) {
        if (DEBUG) {
            Log.d(LOGTAG, args == null || args.length == 0 ? message
                    : String.format(message, args));
        }
    }

    public static void d(String tag, String message, Object... args) {
        if (DEBUG) {
            Log.d(LOGTAG + "/" + tag, args == null || args.length == 0 ? message
                    : String.format(message, args));
        }
    }

    public static void i(String message, Object... args) {
        if (DEBUG) {
            Log.i(LOGTAG, args == null || args.length == 0 ? message
                    : String.format(message, args));
        }
    }

    public static void i(String tag, String message, Object... args) {
        if (DEBUG) {
            Log.i(LOGTAG + "/" + tag, args == null || args.length == 0 ? message
                    : String.format(message, args));
        }
    }

    public static void w(String message, Object... args) {
        if (DEBUG) {
            Log.w(LOGTAG, args == null || args.length == 0 ? message
                    : String.format(message, args));
        }
    }

    public static void w(String tag, String message, Object... args) {
        if (DEBUG) {
            Log.w(LOGTAG + "/" + tag, args == null || args.length == 0 ? message
                    : String.format(message, args));
        }
    }

    public static void e(String message, Object... args) {
        if (DEBUG) {
            Log.e(LOGTAG, args == null || args.length == 0 ? message
                    : String.format(message, args));
        }
    }

    public static void e(String tag, String message, Object... args) {
        if (DEBUG) {
            Log.e(LOGTAG + "/" + tag, args == null || args.length == 0 ? message
                    : String.format(message, args));
        }
    }

    public static void e(String message, Exception e) {
        if (DEBUG) {
            Log.e(LOGTAG, message, e);
        }
    }

    public static void e(String tag, String message, Exception e) {
        if (DEBUG) {
            Log.e(LOGTAG + "/" + tag, message, e);
        }
    }

    public static void wtf(String message, Object... args) {
        if (DEBUG) {
            Log.wtf(LOGTAG, args == null || args.length == 0 ? message
                    : String.format(message, args));
        }
    }

    public static void wtf(String tag, String message, Object... args) {
        if (DEBUG) {
            Log.wtf(LOGTAG + "/" + tag, args == null || args.length == 0 ? message
                    : String.format(message, args));
        }
    }

}

File: utility/src/main/java/com/ellevsoft/utility/LogUtils.kt
package com.ellevsoft.utility

import android.util.Log

object LogUtils {

    private val LOGTAG = "LogUtils"

    // Shows logs on debug version only (Does not appear on the release version)
    private val DEBUG = BuildConfig.DEBUG

    fun v(message: String, vararg args: Any) {
        if (DEBUG) {
            Log.v(LOGTAG, if (args == null || args.size == 0)
                message
            else
                String.format(message, *args))
        }
    }

    fun v(tag: String, message: String, vararg args: Any) {
        if (DEBUG) {
            Log.v("$LOGTAG/$tag", if (args == null || args.size == 0)
                message
            else
                String.format(message, *args))
        }
    }

    fun d(message: String, vararg args: Any) {
        if (DEBUG) {
            Log.d(LOGTAG, if (args == null || args.size == 0)
                message
            else
                String.format(message, *args))
        }
    }

    fun d(tag: String, message: String, vararg args: Any) {
        if (DEBUG) {
            Log.d("$LOGTAG/$tag", if (args == null || args.size == 0)
                message
            else
                String.format(message, *args))
        }
    }

    fun i(message: String, vararg args: Any) {
        if (DEBUG) {
            Log.i(LOGTAG, if (args == null || args.size == 0)
                message
            else
                String.format(message, *args))
        }
    }

    fun i(tag: String, message: String, vararg args: Any) {
        if (DEBUG) {
            Log.i("$LOGTAG/$tag", if (args == null || args.size == 0)
                message
            else
                String.format(message, *args))
        }
    }

    fun w(message: String, vararg args: Any) {
        if (DEBUG) {
            Log.w(LOGTAG, if (args == null || args.size == 0)
                message
            else
                String.format(message, *args))
        }
    }

    fun w(tag: String, message: String, vararg args: Any) {
        if (DEBUG) {
            Log.w("$LOGTAG/$tag", if (args == null || args.size == 0)
                message
            else
                String.format(message, *args))
        }
    }

    fun e(message: String, vararg args: Any) {
        if (DEBUG) {
            Log.e(LOGTAG, if (args == null || args.size == 0)
                message
            else
                String.format(message, *args))
        }
    }

    fun e(tag: String, message: String, vararg args: Any) {
        if (DEBUG) {
            Log.e("$LOGTAG/$tag", if (args == null || args.size == 0)
                message
            else
                String.format(message, *args))
        }
    }

    fun e(message: String, e: Exception) {
        if (DEBUG) {
            Log.e(LOGTAG, message, e)
        }
    }

    fun e(tag: String, message: String, e: Exception) {
        if (DEBUG) {
            Log.e("$LOGTAG/$tag", message, e)
        }
    }

    fun wtf(message: String, vararg args: Any) {
        if (DEBUG) {
            Log.wtf(LOGTAG, if (args == null || args.size == 0)
                message
            else
                String.format(message, *args))
        }
    }

    fun wtf(tag: String, message: String, vararg args: Any) {
        if (DEBUG) {
            Log.wtf("$LOGTAG/$tag", if (args == null || args.size == 0)
                message
            else
                String.format(message, *args))
        }
    }

}
